src/diffusers/training_utils.py

```diff
@@ -298,11 +322,11 @@ def load_state_dict(self, state_dict: dict) -> None:
             if not all(isinstance(p, torch.Tensor) for p in self.shadow_params):
                 raise ValueError("shadow_params must all be Tensors")
 
-        self.collected_params = state_dict.get("collected_params", None)
-        if self.collected_params is not None:
-            if not isinstance(self.collected_params, list):
-                raise ValueError("collected_params must be a list")
-            if not all(isinstance(p, torch.Tensor) for p in self.collected_params):
-                raise ValueError("collected_params must all be Tensors")
-            if len(self.collected_params) != len(self.shadow_params):
-                raise ValueError("collected_params and shadow_params must have the same length")
+        self.temp_stored_params = state_dict.get("temp_stored_params", None)
+        if self.temp_stored_params is not None:
+            if not isinstance(self.temp_stored_params, list):
+                raise ValueError("temp_stored_params must be a list")
+            if not all(isinstance(p, torch.Tensor) for p in self.temp_stored_params):
+                raise ValueError("temp_stored_params must all be Tensors")
+            if len(self.temp_stored_params) != len(self.shadow_params):
+                raise ValueError("temp_stored_params and shadow_params must have the same length")
```

Comment:
Actually, there's no need to save and load these temp params, they are only really necessary during validation and we don't need to keep them after that.

Reply:
Do you suggest deleting these after the validation?

Reply:
Yes, in the `restore` method, once we the params are restored we can delete the temp params and no need to save/load these in the state dict.

Reply:
We can do `del self.temp_stored_params` but then again in the subsequent iterations, that would lead to errors since `restore()` is called in intervals. Am I missing out on anything?

Reply:
`store` and `restore` should always be called in sequence. It doesn't make sense to call `restore` without `store`. So if it leads to an error that's better. And by delete I mean we could simply set it to `None` after restoring is done.

Reply:
Yeah makes sense now! Will proceed now.